### PR TITLE
[MM-42869] Fix dot menu dark mode hover

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
This PR fixes an issue caused buy borrowing styling from `status-dropdown-menu-global-header`.  Fixed by creating a new class specifically for the dot menu component.

For the CSS, I borrowed from the `status-dropdown-menu-global-header` but on only what was needed to format the RHS decorator.
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42869

#### Screenshots
|  before  |  after  |
|----|----| 
| image | image |
| image | image |
  
#### Release Note

```release-note

```

### CODE REVIEW

**Review by jwilander — APPROVED** [review]
LGTM but I'll let Michel comment on the CSS

File: components/dot_menu/dot_menu.scss
Comment (michelengelen) [inline components/dot_menu/dot_menu.scss:10]: NICE ... finally someone that uses `grid`! :D

File: components/dot_menu/dot_menu.scss
Comment (michelengelen) [inline components/dot_menu/dot_menu.scss:1]: I would prefer to name this `dropdown-menu__dotmenu` for a simple reason: We already have the class `dropdown-menu` and this i just a variation of that.

File: components/dot_menu/dot_menu.scss
Comment (michelengelen) [inline components/dot_menu/dot_menu.scss:3]: I don't think we need this here. This might even be prone to bugs when some parent element has a value for `left` defined

File: components/dot_menu/dot_menu.tsx
Comment (michelengelen) [inline components/dot_menu/dot_menu.tsx:434]: according to the comment in `dot_menu.scss`:

```suggestion
                className={'dropdown-menu__dotmenu'}
```

**Review by michelengelen — CHANGES_REQUESTED** [review]
as a side-note, since you did not build this component: the naming is a bit inconsistent. We name the files here `dot_menu.tsx`, which follows the convention of separating pascal case namings with underscores (`DotMenu` -> `dot_menu.tsx`), but the naming in the CSS classes do not.

I am 4/5 that this is out of scope of this PR, but wanted to raise awareness anyways. I hope you don't mind.

**Review by michelengelen — APPROVED** [review]
LGTM

**Comment by jfrerich** [comment]
> as a side-note, since you did not build this component: the naming is a bit inconsistent. We name the files here `dot_menu.tsx`, which follows the convention of separating pascal case namings with underscores (`DotMenu` -> `dot_menu.tsx`), but the naming in the CSS classes do not.
> 
> I am 4/5 that this is out of scope of this PR, but wanted to raise awareness anyways. I hope you don't mind.

Can you give an example of a CSS class that would be renamed in your example.  Perhaps there is an eslint rule that could be created to ensure class names are pascal case?  Perhaps this could be a HW campaign?

**Comment by matthewbirtch** [comment]
Looks good to me @jfrerich. I noticed another issue about the theme color used for menu items with the `dangeours` hover state, but I'll create a separate ticket for that (only in the `Quartz` theme) - screenshot below

image

**Comment by michelengelen** [comment]
> > as a side-note, since you did not build this component: the naming is a bit inconsistent. We name the files here `dot_menu.tsx`, which follows the convention of separating pascal case namings with underscores (`DotMenu` -> `dot_menu.tsx`), but the naming in the CSS classes do not.
> > I am 4/5 that this is out of scope of this PR, but wanted to raise awareness anyways. I hope you don't mind.
> 
> Can you give an example of a CSS class that would be renamed in your example. Perhaps there is an eslint rule that could be created to ensure class names are pascal case? Perhaps this could be a HW campaign?

the CSS class for the `DotMenu` component would be such a case, since `DotMenu` would result in a CSS class `dot-menu`

**Comment by jfrerich** [comment]
Thanks @matthewbirtch, can you mark approved if you don't have any issues outside of your new finding (dangerous hover state)

**Review by matthewbirtch — APPROVED** [review]

**Comment by amyblais** [comment]
@DHaussermann @saturninoabril This one would be nice to get merged soon :)

**Review by saturninoabril — APPROVED** [review]
Thanks @jfrerich! Tested and passed.

**Pull request merged**
